packages/oats-runtime/src/make.ts

```diff
@@ -198,10 +211,31 @@ export function makeString(
   };
 }
 
-export function makeNumber(min?: number, max?: number): Maker<number, number> {
-  return (value: any) => {
-    if (typeof value !== 'number') {
-      return getErrorWithValueMsg('expected a number', value);
+export function makeNumber(min?: number, max?: number) {
+  return makeFloatOrInteger(false, min, max);
+}
+export function makeInteger(min?: number, max?: number) {
+  return makeFloatOrInteger(true, min, max);
+}
+function makeFloatOrInteger(
+  isInteger: boolean,
+  min?: number,
+  max?: number
+): Maker<unknown, number> {
+  return (x, { parseNumericStrings = false } = {}) => {
+    let value: number;
+    if (typeof x === 'number') {
+      value = x;
+    } else if (typeof x === 'string' && parseNumericStrings) {
+      value = parseFloat(x);
```

Comment:
are we concerned that `parseFloat('1a') === 1`

Reply:
perhaps `parseFloat(x) === Number(x)` should be a good enough way to make sure we have valid numbers in decimal format.

Reply:
wouldn't using `value = Number(x)` do the thing without having to deal with parseFloat?